File: InMotionGIT.Common/DataType/LookUpValue.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Runtime.Serialization;
using System.Xml.Serialization;
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.DataType
{

    [DataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
    [XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Common.Attributes.TypeStructureAttribute("Code")]
    public class LookUpValue
    {

        [DataMember()]
        [XmlAttribute()]
        public string Code { get; set; }

        [DataMember()]
        [XmlAttribute()]
        public string Description { get; set; }

        public LookUpValue() : base()
        {
        }

        public LookUpValue(string code)
        {
            Code = code;
        }

        public LookUpValue(string code, string description)
        {
            Code = code;
            Description = description;
        }

        public override string ToString()
        {
            return Description;
        }

        public static explicit operator int(LookUpValue value)
        {
            if (value == null)
            {
                return Conversions.ToInteger(string.Empty);
            }
            else
            {
                return Conversions.ToInteger(value.Code);
            }
        }

        public static implicit operator LookUpValue(string code)
        {
            return new LookUpValue(code);
        }

        public static object LoadFromDataTable(DataTable tableInformation)
        {
            var Result = new List<LookUpValue>();
            LookUpValue item;
            foreach (DataRow row in tableInformation.Rows)
            {
                item = new LookUpValue() { Code = Conversions.ToString(row["CODE"]), Description = Conversions.ToString(row["DESCRIPTION"]) };
                Result.Add(item);
            }
            return Result;
        }

    }

}

File: Common/InMotionGIT.Common/DataType/LookUpValue.vb
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <DataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Attributes.TypeStructure("Code")>
    Public Class LookUpValue

        <DataMember()>
        <XmlAttribute()>
        Public Property LanguageId As Integer

        <DataMember()>
        <XmlAttribute()>
        Public Property LanguagCode As String

        <DataMember()>
        <XmlAttribute()>
        Public Property Code As String

        <DataMember()>
        <XmlAttribute()>
        Public Property Description As String

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal code As String)
            Me.Code = code
        End Sub

        Public Sub New(ByVal code As String, ByVal description As String)
            Me.Code = code
            Me.Description = description
        End Sub

        Public Overrides Function ToString() As String
            Return Description
        End Function

        Public Shared Narrowing Operator CType(ByVal value As LookUpValue) As Integer
            If IsNothing(value) Then
                Return String.Empty
            Else
                Return value.Code
            End If
        End Operator

        Public Shared Widening Operator CType(ByVal code As String) As LookUpValue
            Return New LookUpValue(code)
        End Operator

        Public Shared Function LoadFromDataTable(tableInformation As DataTable)
            Dim Result As New List(Of LookUpValue)
            Dim item As LookUpValue
            For Each row As Data.DataRow In tableInformation.Rows
                item = New LookUpValue With {.Code = row("CODE"), .Description = row("DESCRIPTION")}
                Result.Add(item)
            Next
            Return Result
        End Function

    End Class

End Namespace